STY: updates to export options

jklenzing opened the pull request:
# Description

Updates the pysat export options for operational instruments.  Better integration with https://github.com/jklenzing/ops_reach/pull/15

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

via updated unit tests
```
pytest pysat/tests/test_utils_io.py::TestLoadNetCDF::test_basic_write_and_read_netcdf_export_pysat_info
```

**Test Configuration**:
* Operating system: Mac 13.3.1
* Version number: Python 3.10.8

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

jklenzing commented on CHANGELOG.md:
```diff
@@ -75,6 +75,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
     constellation dataset
   * Reduced default num_samples for constellation test objects
   * Improved consistency in metadata for test instruments
+  * Set upper version limit on `hacking` for flake8 tests
```
```suggestion
```

JonathonMSmith commented on pysat/utils/io.py:
```diff
@@ -1725,7 +1725,8 @@ def inst_to_netcdf(inst, fname, base_instrument=None, epoch_name=None,
     # for including them in the file.
     pysat_items = ['Date_End', 'Date_Start', 'File', 'File_Date',
-                   'Generation_Date', 'Logical_File_ID']
+                   'Generation_Date', 'Logical_File_ID', 'acknowledgements',
```
Wanted to bring this question into the pull request.
Do these items still need to be popped?
We're assigning all of them in the following lines, so maybe they can be left un-popped.

jklenzing replied:
I haven't tried swapping these out.  Not sure if this is a remnant from python 2.7.  acknowledgements and references have to be popped for compatibility with the operational bits below.

jklenzing commented:
@JonathonMSmith, I'd like to get this one into the RC if possible.  What's the best path forward to address the issues above?

jklenzing commented:
This is needed for the operational data stuff for REACH

JonathonMSmith commented:
I don't know the best path forward. I removed the lines where these variables are popped in my local version of the RC branch and all of the unit tests pass, but I don't know if that's enough. I don't see how any of these attributes could get through without being overwritten because the the assignments on lines 1735-1756 are not conditional, the appear to always occur.

jklenzing commented:
> I don't know the best path forward. I removed the lines where these variables are popped in my local version of the RC branch and all of the unit tests pass, but I don't know if that's enough. I don't see how any of these attributes could get through without being overwritten because the the assignments on lines 1735-1756 are not conditional, the appear to always occur.

I've added #1122 for the next version.  We can make a decision then.  The rest of this still needs a review.
